src/porepy/compositional/base.py

```diff
@@ -0,0 +1,855 @@
+
+    The mixture class serves as a container for components and phases and to determine
+    the reference component and phase.
+
+    If also allocates attributes for some thermodynamic properites of a mixture, which
+    are required by the remaining framework, which are assigned by an instance of
+    :class:`~porepy.compositional.compositional_mixins.FluidMixtureMixin`.
+
+    - :attr:`density`
+    - :attr:`specific_enthalpy`
+    - :attr:`specific_volume` as the reciprocal of :attr:`density`
+
+    The mixture allows only one gas-like phase, and it must be modelled with at least
+    1 component and 1 phase.
+
+    Flash algorithms are built around the mixture management utilities of this class.
+
+    Important:
+        Phases are re-ordered once passed as arguments.
+
+        - If more than 1 phase, the first, non-gas-like phase is treated as the
+          reference phase.
+        - The single gas-like phase is always the last one.
+        - The first component is set as reference component.
+
+    Notes:
+        Be careful when modelling mixtures with 1 component. This singular case is not
+        fully supported by the equilibrium framework.
+
+    Parameters:
+        components: A list of components to be added to the mixture.
+            This are the chemical species which can appear in multiple phases.
+        phases: A list of phases to be modelled.
+
+    Raises:
+        CompositionalModellingError: If the model assumptions are violated.
+
+            - at most 1 gas phase must be modelled.
+            - At least 1 component must be present.
+            - At least 1 phase must be modelled.
+            - Any phase has no components in it.
+        CompositionalModellingError: If there is 1 component, which is not in any phase.
+        ValueError: If any two components or phases have the same name
+            (storage conflicts).
+
+    """
+
+    def __init__(
+        self,
+        components: list[Component],
+        phases: list[Phase],
+    ) -> None:
+        # modelled phases and components
+        self._components: list[Component] = []
+        """A list of components passed at instantiation."""
+        self._phases: list[Phase] = []
+        """A list of phases passed at instantiation."""
+
+        # a container holding names already added, to avoid storage conflicts
+        doubles = []
+        # Lists of gas-like and other phases
+        gaslike_phases: list[Phase] = list()
+        other_phases: list[Phase] = list()
+
+        for comp in components:
+            if comp.name in doubles:
+                raise ValueError(
+                    f"All components must have different 'name' attributes."
+                )
+            doubles.append(comp.name)
+            self._components.append(comp)
+
+        for phase in phases:
+            if phase.name in doubles:
+                raise ValueError("All phases must have different 'name' attributes.")
+            doubles.append(phase.name)
```

Comment:
Yes, I prefer the chemical formula for components, especially since the look up in e.g. ``chemicals`` is more robust using the formula.
And then a meaningful name for phases (oil, aquatic, water, gas, liquid)